dev.py

```diff
@@ -409,7 +409,12 @@ def setup_build(args, env):
     """
     cmd = ["meson", "setup", args.build_dir, "--prefix", PATH_INSTALLED]
     build_dir = Path(args.build_dir)
-    if os.path.exists(build_dir / 'build.ninja'):
+    if (os.path.exists(build_dir) and not os.path.exists(build_dir
+                                                         / 'meson-info')):
+        if len(os.listdir(build_dir)):
+            raise RuntimeError("Can't build into non-empty directory "
+                               f"'{str(build_dir.absolute())}'")
```

Comment:
I think `str` is the default already:

```suggestion
                               f"'{build_dir.absolute()}'")
```